libkbfs/interfaces.go

```diff
@@ -1325,6 +1325,22 @@ type Config interface {
 	// setting the rekey bit, before prompting for a paper key.
 	RekeyWithPromptWaitTime() time.Duration
 
+	// GracePeriod specifies a grace period for which a delayed cancellation
+	// waits before actual cancels the context. This is useful for giving
+	// critical portion of a slow remote operation some extra time to finish as
+	// an effort to avoid conflicting. Example include an O_EXCL Create call
+	// interrupted by ALRM signal actually makes it to the server, while
+	// application assumes not since EINTR is returned. A delayed cancellation
+	// allows us to dinstinct between successful cancel (where remote operation
+	// didn't make to server) or failed cancel (where remote operation made to
+	// the server). However, the optimal value of this depends on the network
+	// conditions. A long grace period for really good network condition would
+	// just unnecessarily slow down Ctrl-C.AuthTokenMinRefreshSeconds
+	//
+	// TODO: make this adaptive and self-change over time based on network
+	// conditions.
+	GracePeriod() time.Duration
+	SetGracePeriod(time.Duration)
```

Comment:
Also i think you should run `./gen_mocks.sh` to pick up these changes in the (not very useful) `MockConfig` definition.